### DISCUSSION TITLE
Order creates a problem of 422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by DScabezas** [fastapi/fastapi#13060] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List
from sqlmodel import select
from fastapi import APIRouter, status
from app.db import SessionDep
from app.models.kpis import Kpi

router = APIRouter()

# Route that fails if it's defined before the other
@router.get(
    "/kpis/active",
    response_model=int,
    status_code=status.HTTP_200_OK,
    tags=["KPIs"],
)
def count_all_kpis(session: SessionDep):
    """
    Fetches the total number of KPIs available.
    """
    kpis = session.exec(select(Kpi)).all()
    return len(kpis)


# Route that works correctly
@router.get(
    "/kpis",
    response_model=List[Kpi],
    status_code=status.HTTP_200_OK,
    tags=["KPIs"],
)
def get_all_kpis(session: SessionDep):
    """
    Fetches all available KPIs.
    """
    kpis = session.exec(select(Kpi)).all()
    return kpis
```

### Description

An error 422 (Unprocessable Entity) occurs when making a request to the route GET /kpis/active if the function get_all_kpis (which fetches all KPIs) is defined first in the code. However, if the GET /kpis/active route is defined before the GET /kpis route, the request works correctly without the error.

INFO:     127.0.0.1:59134 - "GET /kpis/active HTTP/1.1" 422 Unprocessable Entity

Cause of the Error: The error appears to be related to the order in which the routes are defined in the code. When the GET /kpis/active route is defined before GET /kpis, the error occurs. However, when the positions of the routes are swapped, the issue disappears.

Expected Behavior: Both routes should work correctly regardless of the order in which they are defined in the code.

### Operating System

Linux, Windows

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

Python 3.11.9

### Additional Context

Model used

from sqlmodel import SQLModel, Field, Relationship
from typing import List, Optional

from app.models.catalogs import Catalog
from app.models.records import Records
from app.models.utils import Chart, Color

class KpiBase(SQLModel):
    name: str
    description: Optional[str] = None
    formula: Optional[str] = None
    position_index: Optional[int] = Field(default=None)

class Kpi(KpiBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    catalog_id: Optional[int] = Field(foreign_key="catalog.id")
    color_schema: Optional[int] = Field(foreign_key="color.id")
    chart_type: Optional[int] = Field(foreign_key="chart.id")
    catalog: Optional["Catalog"] = Relationship(back_populates="kpis")
    color: Optional["Color"] = Relationship(back_populates="kpi")
    chart: Optional["Chart"] = Relationship(back_populates="kpi")
    records: List["Records"] = Relationship(back_populates="kpi")

**Comment by luzzodev** [comment]
Hello @DScabezas the order in your example doesn't matter, i have tested it and works in both directions as expected.
Are you sure you have posted a proper example of your issue?
422 Validation is raised from fastapi in the solving dep step, and in your example the are no dependencies to solve except for the db connection that does not depends from any external input.

Could you add the content of your 422?
Also i have check at your project and i do not see any `count_all_kpis` function.

If you haven't pushed yet then the conflict could happen with this function
``` python

@router.get(
    "/kpis/{kpi_id}",
    response_model=Kpi,
    status_code=status.HTTP_200_OK,
    tags=["KPIs"],
)
def get_kpi(kpi_id: int, session: SessionDep):
    """
    Obtiene un KPI por su ID.
    """
    kpi = session.exec(select(Kpi).where(Kpi.id == kpi_id)).first()
    if not kpi:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND, detail="KPI not found"
        )
    return kpi
```

And in this case order matters.
